.circleci/config.yml

```diff
@@ -520,21 +518,45 @@ workflows:
     jobs:
       - merge_pr_to_develop
 
-  nightly_build:
+  # Python versions that are supported on `main`.
+  nightly_build_main:
     triggers:
       - schedule:
           cron: 30 2 * * *
           filters:
             branches:
               only: main
     jobs:
-      - build_docker_image
+      - build_docker_image:
+          matrix:
+            parameters:
+              python_version: ["3.6", "3.7", "3.8"]
       - build_kedro:
           matrix:
             parameters:
               python_version: ["3.6", "3.7", "3.8"]
           requires:
-            - build_docker_image
+            - build_docker_image-<<matrix.python_version>>
```

Comment:
This means that `build_kedro-3.6` only depends on `build_docker_image-3.6`, etc.